### ISSUE TITLE
v4: hard-coded border-width properties

### ISSUE THREAD

**Issue by davidjb** [twbs/bootstrap#18150]
Looking through v4-dev, the `$border-width` variable is available, but only sporadically used (list-groups, navbar, hr, etc).  There's a variety of different places `1px` borders are hard-coded (dropdowns, nav, model, images, tables) and then components such as `nav` use a negative margin for positioning over the border.

Would a PR that changes these `1px` references over the `$border-width` be acceptable, or are there particular components that require precisely `1px`?  Alternatively, should components each have their own configurable borders and just default to $border-width (ala #17939)?

**Comment by cvrebert** [comment]
> Alternatively, should components each have their own configurable borders and just default to $border-width (ala #17939)?

Since that PR has been merged by @mdo, I think that approach is preferred. It's more flexible/customizable than the other option.

---

As of 6d591edcb7540eff05b21ef53205aa8517080432, the offending borders seem to be:
```
scss/_card.scss:123:    border-bottom: .075rem solid rgba(255,255,255,.2);
scss/_custom-forms.scss:141:  border: 1px solid $input-border;
scss/_custom-forms.scss:198:  border: .075rem solid #ddd;
scss/_custom-forms.scss:218:  border: .075rem solid #ddd;
scss/_dropdown.scss:54:  border: 1px solid $dropdown-border;
scss/_images.scss:22:  border: 1px solid $thumbnail-border;
scss/_nav.scss:62:    border: 1px solid transparent;
scss/_pager.scss:17:      border: 1px solid $pager-border;
scss/_pagination.scss:21:      border: 1px solid $pagination-border;
scss/_popover.scss:110:  border-bottom: 1px solid darken($popover-title-bg, 5%);
scss/_print.scss:27:    border: 1px solid #999;
scss/_print.scss:69:    border: 1px solid #000;
scss/_print.scss:83:      border: 1px solid #ddd !important;
scss/_tables.scss:60:      border-bottom-width: 2px;
```

**Comment by mdo** [comment]
I imagine we could use `$border-width` for all these, but I'm wondering if we should use reassigned variables instead? For example:

```scss
$card-border-width: $border-width !default;
```

**Comment by davidjb** [comment]
+1 for @mdo's suggestion of reassigned variables; that'd be extremely flexible for different applications and themes.

**Comment by cvrebert** [comment]
@mdo Is it okay to change those `.075rem`-s in `_custom-forms.scss` to (variables which default to) `$border-width` (i.e. `1px`) ?

**Comment by cvrebert** [comment]
This should be fully fixed after #18380 & #18381 are merged.

**Comment by mdo** [comment]
Done!
